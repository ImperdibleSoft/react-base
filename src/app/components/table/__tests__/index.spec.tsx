import * as React from 'react';
import { shallow } from 'enzyme';

import { Table, THead, TBody, Tr, Th, Td } from '..';

describe('Table', () => {
  it('should render the complete set of table components', () => {
    const component = shallow(
      <Table>
        <THead>
          <Tr>
            <Th>Title</Th>
          </Tr>
        </THead>
        <TBody>
          <Tr>
            <Td>Row 1</Td>
          </Tr>
          <Tr>
            <Td>Row 2</Td>
          </Tr>
        </TBody>
      </Table>
    );
    expect(component).toMatchSnapshot();
  });
});
