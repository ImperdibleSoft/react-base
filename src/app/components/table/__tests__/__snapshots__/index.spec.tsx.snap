// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render the complete set of table components 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "styles__Table-sc-1qcs71b-0",
        "isStatic": false,
        "rules": Array [
          "border-collapse:collapse;margin-bottom:10px;width:100%;",
        ],
      },
      "displayName": "Table",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "styles__Table-sc-1qcs71b-0",
      "target": "table",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <THead>
    <Tr>
      <Th>
        Title
      </Th>
    </Tr>
  </THead>
  <TBody>
    <Tr>
      <Td>
        Row 1
      </Td>
    </Tr>
    <Tr>
      <Td>
        Row 2
      </Td>
    </Tr>
  </TBody>
</StyledComponent>
`;
